### PR TITLE
[GAPRINDASHVILI] Updated service template's ui token info check to use token metadata for requester_type

### PR DESCRIPTION
**Pull request by gtanzillo**
Updated service template's ui token info check to use token metadata for requester_type

Cherry-pick of pull request #529 from abellotti/service_template_ui_token_fix
(cherry picked from commit cfcb10b912d5a8b2b01f96e956d3d790ce5a1818)

Manually fixed conflicts

https://bugzilla.redhat.com/show_bug.cgi?id=1660237

### CODE REVIEW

File: lib/services/api/user_token_service.rb [hunk 1]
```diff
@@ -32,9 +32,9 @@ def generate_token(userid, requester_type)
 
       # Additional Requester type token ttl's for authentication
       type_to_ttl_override = {'ui' => ::Settings.session.timeout}
-
-      token_mgr(requester_type).gen_token(:userid             => userid,
-                                          :token_ttl_override => type_to_ttl_override[requester_type])
+      token_metadata = { :userid => userid, :token_ttl_override => type_to_ttl_override[requester_type] }
```
Comment (gtanzillo) [inline lib/services/api/user_token_service.rb]: @abellotti This is where the only conflict was. I made the same change (passing `type_to_ttl_override[requester_type]`) as we did on the appliance.

**Comment by simaishi** [comment]
@gtanzillo Please add BZ link. Thanks!

**Comment by gtanzillo** [comment]
@simaishi Here's the BZ https://bugzilla.redhat.com/show_bug.cgi?id=1659602. That's the original one, not sure if it needs a clone

**Comment by simaishi** [comment]
Created clone: https://bugzilla.redhat.com/show_bug.cgi?id=1660237

File: lib/services/api/user_token_service.rb (hunk 1 above)
Comment (abellotti) [reply lib/services/api/user_token_service.rb]: yep makes sense, gotta get the right expected metadata for this release.

**Pull request merged**